src/Tempest/Http/Session/Managers/FileSessionManager.php

```diff
@@ -114,8 +124,8 @@ private function persist(SessionId $id, ?array $data = null): Session
         $path = $this->getPath($id);
         $dir = dirname($path);
 
-        if (! is_dir($dir)) {
-            mkdir($dir, recursive: true);
+        if (! is_dir($dir) && ! mkdir($dir, recursive: true) && ! is_dir($dir)) {
+            throw new RuntimeException(sprintf('Directory "%s" was not created', $dir));
```

Comment:
Not sure if this was Rector or the PHP Inspections (EA extended) from my IDE.

But it adds an extra check if the directory can actually be created.

Reply:
Hmm I don't like these complex `if` statements. Let's revert this one. I plan on adding a proper path helper at some point which will take care of this.